transport/socks5/stream_dialer.go

```diff
@@ -65,15 +68,14 @@ func (c *StreamDialer) SetCredentials(username, password []byte) error {
 	return nil
 }
 
-// DialStream implements [transport.StreamDialer].DialStream using SOCKS5.
-// It will send the auth method, auth credentials (if auth is chosen), and
-// the connect requests in one packet, to avoid an additional roundtrip.
-// The returned [error] will be of type [ReplyCode] if the server sends a SOCKS error reply code, which
-// you can check against the error constants in this package using [errors.Is].
-func (c *StreamDialer) DialStream(ctx context.Context, remoteAddr string) (transport.StreamConn, error) {
-	proxyConn, err := c.proxyEndpoint.ConnectStream(ctx)
+func (c *Client) EnablePacket(packetDialer transport.PacketDialer) {
+	c.pd = packetDialer
+}
+
+func (c *Client) request(ctx context.Context, cmd byte, dstAddr string) (transport.StreamConn, string, error) {
```

Comment:
This function is doing a bit too much. Can you make it take an io.ReadWriter and move the connection to the caller instead? This way we simplify the return value and separate the concerns.

Reply:
@fortuna good point. I refactored the code accordingly.